### PR TITLE
Implement class atomic<double> and atomic<float>

### PR DESCRIPTION
**Pull request by st-pasha**
`std::atomic<double>` is specialized only in C++17, not in C++11. In C++11 there is generic `std::atomic<T>`, but it lacks arithmetic operators, which are very important for our use case. Hence, we implement our own atomic class, which reuses atomic ints at bit level.

WIP for #1736

### CODE REVIEW

File: c/models/dt_ftrl_real.cc [hunk 1]
```diff
@@ -388,20 +388,19 @@ double FtrlReal<T>::fit(T(*linkfn)(T), T(*lossfn)(T,U)) {
               }
             }
           }
-          #pragma omp atomic
           loss += loss_local;
```
Comment (oleksiyskononenko) [inline c/models/dt_ftrl_real.cc]: With your implementation of `dt::atomic`, will it do `loss += loss_local` atomically or we need to use some `loss.operation()` here?
